### PR TITLE
imagebuildah,multi-stage: do not remove base-image when built without `--layers`

### PR DESCRIPTION
**Pull request by flouthoc**
When building a multi-stage image ( without `--layers` ) and a stage contains only a base-image buildah removes the base-image itself as part of cleanup process. This is a bug and following commit fixes that.

Reproducer

```Dockerfile
FROM parent
FROM another-base
COPY --from=0 somefile .
```

`buildah build -t multi-stage .`

Closes: https://github.com/containers/podman/issues/20291

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
imagebuildah,multi-stage: do not remove base-image when build without `--layers`
```

### CODE REVIEW

**Comment by flouthoc** [comment]
@containers/buildah-maintainers PTAL, 

@lsm5 it seems `rpm-build` also blocks merge bot.

**Comment by lsm5** [comment]
> @containers/buildah-maintainers PTAL,
> 
> @lsm5 it seems `rpm-build` also blocks merge bot.

I'm checking on the ELN issue, but right now this PR doesn't yet have the lgtm label, so I'm not sure if it's actually blocked on that. Do you see mergebot blocked on ELN failures in other PRs with lgtm label already?

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@lsm5 @nalind @giuseppe @umohnani8 @vrothberg PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by flouthoc** [comment]
@rhatdan I don't think this will be merged, since `rpm-build` is failing so merge bot is blocked, I had merge some previous PR's manually. cc @lsm5

**Comment by lsm5** [comment]
ok, this time i'm seeing the `Not mergeable. Job rpm-build:fedora-eln-aarch64 has not succeeded.` Let me check again in repo settings.

**Comment by lsm5** [comment]
> ok, this time i'm seeing the `Not mergeable. Job rpm-build:fedora-eln-aarch64 has not succeeded.` Let me check again in repo settings.

ok, in Settings -> Branches, for the main repo Success task, I changed the value from `Any source` to `Cirrus CI`. I think going forward, jobs shouldn't be blocked on rpm build. But this one might need a manual merge.

**Comment by nalind** [comment]
Tide can be told to ignore specific status checks with its "override" command.

**Pull request merged**
